data/src/main/scala/ch.epfl.scala.index.data/download/PlayWsDownloader.scala

```diff
@@ -186,18 +184,19 @@ trait PlayWsDownloader {
       }
     }
 
-    processDownloads(message, toDownload, processItem, parallelism)
+    processDownloads(message, toDownload, processItem)
   }
 
   private def processDownloads[T, R](
       message: String,
       toDownload: Set[T],
-      processItem: (AhcWSClient, T, ProgressBar) => Future[R],
-      parallelism: Int
+      processItem: (AhcWSClient, T, ProgressBar) => Future[R]
   ): Seq[R] = {
 
     def processItems(client: AhcWSClient, progress: ProgressBar) = {
-
+      // use minimal concurrency to avoid abuse rate limit error which is triggered
+      // by making too many calls in a short period of time, see https://github.com/scalacenter/scaladex/issues/431
+      val parallelism = 1
       Source(toDownload).mapAsyncUnordered(parallelism) { item =>
```

Comment:
you can use named parameters here: `mapAsyncUnordered(parallelism = 1)`

Can we set it to 4? We make all the effort to have parallel code but set it sequentiail :S. There is probably a sweet spot, What kind of rate limit stas Github give us?